### PR TITLE
[SPARK-17087][MESOS] Documentation for Making Spark on Mesos honor port restrictions

### PR DESCRIPTION
**Pull request by skonto**
## What changes were proposed in this pull request?
- adds documentation for https://issues.apache.org/jira/browse/SPARK-11714
## How was this patch tested?

Doc no test needed.

### CODE REVIEW

File: docs/running-on-mesos.md [hunk 1]
```diff
@@ -207,6 +207,16 @@ The scheduler will start executors round-robin on the offers Mesos
 gives it, but there are no spread guarantees, as Mesos does not
 provide such guarantees on the offer stream.
 
+In this mode spark executors will honour port allocation if such is
```
Comment (srowen) [inline docs/running-on-mesos.md]: OK by me, even in the UK English spelling.

Comment (skonto) [inline docs/running-on-mesos.md]: @srowen  Let me know how you want it rephrased and I will update it, np.

Comment (srowen) [inline docs/running-on-mesos.md]: Yeah, probably not worth worrying about unless you have a spare minute... I suppose the docs consistently use US spelling but it's hardly a big deal.

**Comment by skonto** [comment]
Ok I will fix it in a second.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**